pipeline/frontend/yaml/linter/linter_test.go

```diff
@@ -172,3 +181,41 @@ func TestLintErrors(t *testing.T) {
 		assert.True(t, found, "Expected error %q, got %q", test.want, lerrors)
 	}
 }
+
+func TestBadHabits(t *testing.T) {
+	testdata := []struct {
+		from string
+		want string
+	}{
+		{
+			from: "steps: { build: { image: golang } }",
+			want: "Please set an event filter",
+		},
+		{
+			from: "when: [{branch: xyz}, {event: push}]\nsteps: { build: { image: golang } }",
+			want: "Please set an event filter",
```

Comment:
why does it suggest an event filter if global does already have it?

Reply:
Well i guess, user having questions abour that message or not will finaly tell us